### ISSUE TITLE
v1 API Support

### ISSUE THREAD

**Issue by ataylorme** [UpThemes/freefolio#4]
> Dribbble v1 API: Beta Than Ever
> Currently in beta and subject to change, but we hope to mark it as a stable release soon.
> The original Dribbble API is still available but will be retired on April 8, 2015.
> [source](http://developer.dribbble.com/v1/)

I created a branch [new-api](https://github.com/UpThemes/freefolio/tree/new-api) for testing v1 of the Dribbble API and have existing functionality working with the new API

Looking at adding a toggle to automatically grab new Dribble shots after the initial import it run.

This would probably use [wp_schedule_event](http://codex.wordpress.org/Function_Reference/wp_schedule_event) to trigger a function that requests all shots for the current day using [the date parameter of the new API](http://developer.dribbble.com/v1/shots/#list-shots).

**Comment by ataylorme** [comment]
Did some more cleanup.
- Switched from `get_posts` to `WP_Query`
- Switched from `file_get_contents` to [`wp_remote_get`](http://codex.wordpress.org/Function_Reference/wp_remote_get)
- Made API key an option rather than hard-coding
- Removed `wp_unschedule_event` after initial import; Import now runs hourly

**Comment by issuepressapp** [comment]
Did you push this to SVN yet?

On Mon, Dec 29, 2014 at 9:03 AM, Andrew Taylor notifications@github.com
wrote:

> Did some more cleanup.
> - Switched from get_posts to WP_Query
> - Switched from file_get_contents to wp_remote_get
>   http://codex.wordpress.org/Function_Reference/wp_remote_get
> - Made API key an option rather than hard-coding
> - Removed wp_unschedule_event after initial import; Import now runs
>   hourly
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/UpThemes/freefolio/issues/4#issuecomment-68264329.

## 

Chris Wallace
Principal Creative // Lift

mobile 314-406-9109
skype chriswallace7
email/gtalk chris@liftux.com

http://liftux.com

**Comment by ataylorme** [comment]
No, just a Git branch for now. The new Dribbble API is still in Beta and subject to change. I set a milestone on this for April, when the current API will be deprecated. We can merge prior if the new API comes out of Beta.

**Comment by christianc1** [comment]
The old api is new deprecated.  The plugin in .org plugin repo is broken.  I test the new-api branch, everything appears to work as expected with one exception.  Without jetpack installed, portfolio cpt isn't created.

see:https://github.com/UpThemes/freefolio/blob/a98135af417391e8907734c08c857efa14cd9ff7/class.Freefolio.php#L81

Seems counter-intuitive to me, but perhaps it's because I was just doing a quick code review.
